### ISSUE TITLE
can't format let chains

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rustfmt#5456]
let chains got stabilized recently but it seems that rustfmt has absolutely no idea how to format them https://github.com/rust-lang/rust/pull/94927

sample code, nothing will be reformatted in this code:
````rust
pub enum Color {
    Blue,
    Red,
    Violet,
}

pub enum Flower {
    Rose,
    Tulip,
    Violet,
}

pub fn roses_are_red_violets_are_blue_printer(
    (first_flower, first_flower_color): (Flower, Color),
    (second_flower, second_flower_color): (Flower, Color),
    pick_up_lines: &[&str],
) {
	// I messed up the formatting here a bit
    if let Flower::Rose = first_flower
        && let Color::Red = first_flower_color &&
      let Flower::Violet = second_flower
        && let Color::Blue =
        second_flower_color
        && let &[first_pick_up_line, ..] = pick_up_lines
      {
 println!("Roses are red, violets are blue, {}", first_pick_up_line);
    }
}

fn main() {
    roses_are_red_violets_are_blue_printer(
        (Flower::Rose, Color::Red),
        (Flower::Violet, Color::Blue),
        &["sugar is sweet and so are you"],
    );
}

````

**Comment by calebcartwright** [comment]
Thanks but going to close as a duplicate of #4955.

tl;dr, rustfmt doesn't format let expressions because there isn't any specification yet for _how_ they should be formatted, even more so in chains - https://github.com/rust-dev-tools/fmt-rfcs/issues/169
